ntbs-service/Pages/Notifications/Edit/Patient.cshtml.cs

```diff
@@ -80,25 +90,9 @@ private async Task FindAndSetPostcodeAsync()
             var foundPostcode = await PostcodeService.FindPostcode(Patient.Postcode);
             Patient.PostcodeToLookup = foundPostcode?.Postcode;
         }
-        
-        private void UpdatePatientFlags()
-        {
-            if (Patient.NhsNumberNotKnown)
-            {
-                Patient.NhsNumber = null;
-                ModelState.Remove("Patient.NhsNumber");
```

Comment:
I'm worried that by no longer modifying the model state, we lose the ability to cope with the following scenario:
1. click nhs number known
1. enter invalid number
1. submit form - see errors
1. click nhs number not known
1. submit form successfully

The initially attempted nhs number will still be passed through (coz the field is just hidden), and a validation will kick in and fail

Am I missing something that deals with this?

Reply:
aaaaand now I see what I was missing. All good here :)

Reply:
Nah you were right the first time 
I neglected to clearValidationState for NHS Number (only postcode and my new field)

Reply:
aha, double bluff! :D